src/Microsoft.DotNet.Build.Tasks/PackageFiles/publishtest.targets

```diff
@@ -150,12 +153,12 @@
       </_SupplementalTestData>
     </ItemGroup>
     <ItemGroup>
-      <SupplementalTestDataTestDir Include="@(_SupplementalTestData->'$(TestPath)/%(DestinationDir)%(Filename)%(Extension)')" />
```

Comment:
True; it looks like all of the files copied in CopyTestToTestDirectory are already in OutDir, so copying them there again would just no-op.  However, I think that would make the code a bit confusing, to see the test binaries get copied to the OutDir in the test targets even though they're already there from the build targets.

Another difference is that the supplemental files typically go in subdirectories of TestPath and OutDir, whereas the SourcesToCopyToTestDir files typically go directly in TestPath.

These are different types of files that we handle differently, so I think it makes sense to copy them in separate targets.  It makes the code more readable to have the different logical pieces in different targets.